dashboard/modules/job/job_manager.py

```diff
@@ -374,60 +393,53 @@ async def run(
             )
             log_path = self._log_client.get_log_file_path(self._job_id)
             child_process = self._exec_entrypoint(log_path)
+            child_pid = child_process.pid
 
             polling_task = create_task(self._polling(child_process))
             finished, _ = await asyncio.wait(
                 [polling_task, self._stop_event.wait()], return_when=FIRST_COMPLETED
             )
 
             if self._stop_event.is_set():
+                polling_task.cancel()
```

Comment:
Pulling this out of the `if` statement means it'll run on both Linux and MacOS. Is that expected? The PR description says we want this changed on Linux specifically, which is why I'm asking.

Reply:
Sorry for the confusion, these changes will be running on both Linux and MacOS systems, although it was meant to solve a problem specific to Linux systems. As for the `if` statement, it was originally outside before [this PR](https://github.com/ray-project/ray/pull/30851), where I moved it because I wanted to keep using the same polling task. However with this new PR, I want to change to a new polling task that polls all processes, so I moved it back.

Reply:
Ah, I see. Thanks for clarifying!